src/agent0/hyperfuzz/system_fuzz/invariant_checks.py

```diff
@@ -390,46 +388,59 @@ def _check_lp_share_price(
     exception_data: dict[str, Any] = {}
     log_level = None
 
-    block_number = pool_state.block_number
+    # TODO we hack in a stateful variable into the interface here, since we need
+    # to check between subsequent calls here.
+    # Initial call, we look to see if the attribute exists
+    previous_pool_state: PoolState | None = getattr(interface, "_lp_share_price_check_state", None)
+    # Always set the new pending state here
+    setattr(interface, "_lp_share_price_check_state", interface.get_hyperdrive_state("pending"))
+
+    if previous_pool_state is None:
+        # Skip this check on initial call, not a failure
+        return InvariantCheckResults(
+            failed=False, exception_message=exception_message, exception_data=exception_data, log_level=log_level
+        )
+
+    # This is the block we're checking the lp share price on
+    check_block_number = previous_pool_state.block_number
+    # Get the pool state after it was mined
+    mined_pool_state = interface.get_hyperdrive_state(block_data=interface.get_block(check_block_number))
 
     # We expect the lp share price to be less than the test epsilon between sequential blocks
     # However, when simulating, we can advance time by any amount of time. Hence, we define
     # the test epsilon to be relative to 12 seconds (1 block), and normalize by the actual time
     # between blocks.
 
-    # This is known to fail when checking the first block, as block - 1 doesn't exist.
-    try:
-        previous_pool_state = interface.get_hyperdrive_state(block_number - 1)
-    except BlockNotFound:
-        # Not a failure
-        return InvariantCheckResults(
-            failed=False, exception_message=exception_message, exception_data=exception_data, log_level=log_level
-        )
+    # The mined pool state is always 12 seconds ahead of the previous pool state.
```

Comment:
is it always 12 sec when we're fuzzing with a `--block-time=12` flag set for anvil, right? But on mainnet & testnet it's not always 12. Is that going to matter?

Reply:
Resolved via flag when running in simulation mode